server/src/operators/dataset_operator.rs

```diff
@@ -570,7 +570,13 @@ pub async fn update_dataset_query(
             .filter(datasets_columns::deleted.eq(0)),
     )
     .set((
-        new_tracking_id.map(|id| datasets_columns::tracking_id.eq(id)),
+        new_tracking_id.map(|id| {
+            if id == "" {
+                datasets_columns::tracking_id.eq(None)
+            } else {
+                datasets_columns::tracking_id.eq(Some(id))
+            }
+        }),
```

Comment:
Added in a backend part to this that sets an empty tracking_id string to just delete instead, no we don't have to worry about empty states.